Source/WebKit/UIProcess/OpenedPageProxy.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "MessageReceiver.h"
+#include "MessageSender.h"
+#include <WebCore/PageIdentifier.h>
+
+namespace IPC {
+class Connection;
+class Decoder;
+class Encoder;
+}
+
+namespace WebKit {
+
+class ProvisionalPageProxy;
+class WebFrameProxy;
+class WebPageProxy;
+class WebProcessProxy;
+
+class OpenedPageProxy : public IPC::MessageReceiver, public IPC::MessageSender {
```

Comment:
There is something fishy about the design here. Why do we need an OpenedPageProxy?

When we create a popup, we get a regular WebPageProxy. Then, when the initial navigation is cross-origin, we'll create a ProvisionalPageProxy for the process-swap until the load is committed. We should not need a new `OpenedPageProxy` type.

Reply:
Somewhat analogous to SubframePageProxy to track the remote page in another process. Though as it stands now, it's just light sugar over constructing and sending the CreateWebPage message. I'm fine copy pastaing the contents of OpenedPageProxy's constructor to initializeWebPage for now.

Reply:
`SubframePageProxy` is a bit weird too but I believe this was needed because frames in remote processes needed a WebPage on the WebProcess side and these needed an object to communicate to on the UIProcess side. It couldn't be the WebPageProxy because it is already connected to the WebPage associated with the main frame process.

In your case though, it is a popup. A popup gets a brand new WebPageProxy and this WebPageProxy is associated with a single WebProcess (which may change on navigation). This is just regular process-swap-on-navigation and I stand by my assessment that we shouldn't need a new proxy object on the UIProcess side. Why can't we use that WebPageProxy, why would we need yet another proxy object?

So far, your comments didn't convince me we NEED a new proxy object.